examples/esp32/esp32_static.rs

```diff
@@ -0,0 +1,180 @@
+#[allow(dead_code)]
+#[cfg(not(feature = "qemu"))]
+const SSID: &str = env!("MICRO_RDK_WIFI_SSID");
+#[allow(dead_code)]
+#[cfg(not(feature = "qemu"))]
+const PASS: &str = env!("MICRO_RDK_WIFI_PASSWORD");
+
+// Generated robot config during build process
+include!(concat!(env!("OUT_DIR"), "/robot_secret.rs"));
+include!(concat!(env!("OUT_DIR"), "/robot_config.rs"));
+
+#[cfg(all(not(feature = "qemu"), feature = "camera"))]
+use crate::camera::Esp32Camera;
+
+use anyhow::bail;
+use esp_idf_hal::prelude::Peripherals;
+#[cfg(feature = "qemu")]
+use esp_idf_svc::eth::*;
+#[cfg(feature = "qemu")]
+use esp_idf_svc::eth::{EspEth, EthWait};
+use esp_idf_svc::eventloop::EspSystemEventLoop;
+use esp_idf_svc::netif::{EspNetif, EspNetifWait};
+#[cfg(not(feature = "qemu"))]
+use esp_idf_svc::wifi::EspWifi;
+use esp_idf_sys as _; // If using the `binstart` feature of `esp-idf-sys`, always keep this module imported
+#[cfg(not(feature = "qemu"))]
+use esp_idf_sys::esp_wifi_set_ps;
+use log::*;
+use micro_rdk::common::config::{Kind, RobotConfigStatic, StaticComponentConfig};
+use micro_rdk::common::robot::LocalRobot;
+use micro_rdk::common::robot::ResourceType;
+use micro_rdk::esp32::server::{CloudConfig, Esp32Server};
+use micro_rdk::esp32::tls::Esp32TlsServerConfig;
+use micro_rdk::proto::common::v1::ResourceName;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::rc::Rc;
+use std::sync::Arc;
+use std::sync::Mutex;
+use std::time::Duration;
+
+fn main() -> anyhow::Result<()> {
+    esp_idf_sys::link_patches();
+
+    esp_idf_svc::log::EspLogger::initialize_default();
+    let sys_loop_stack = EspSystemEventLoop::take().unwrap();
+    let periph = Peripherals::take().unwrap();
+
+    let robot = LocalRobot::new_from_static(&STATIC_ROBOT_CONFIG.unwrap()).unwrap();
```

Comment:
Neat. At first I was puzzled why there was so much activity in this example, but I see now that it is basically the same as the existing `esp32.rs`, with the whole static config replaced with this call.

It might be good at some point to move some of the common logic needed to get an esp32 up and running into some helpers, rather than needing to duplicate it among all the examples (and, presumably, the project template).

Reply:
once the static config is done and stable we are going to remove the other example and change this one to an upload-able payload.